disnake/guild_scheduled_event.py

```diff
@@ -0,0 +1,372 @@
+# The MIT License (MIT)
+
+# Copyright (c) 2021-present DisnakeDev
+
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, Dict, List, Optional, TYPE_CHECKING, Union
+from .enums import (
+    GuildScheduledEventEntityType,
+    GuildScheduledEventStatus,
+    StagePrivacyLevel,
+    try_enum,
+)
+from .user import User
+from .member import Member
+from .mixins import Hashable
+from .utils import cached_slot_property, parse_time, _get_as_snowflake, MISSING
+
+if TYPE_CHECKING:
+    from .abc import GuildChannel
+    from .guild import Guild
+    from .state import ConnectionState
+
+
+__all__ = ("GuildScheduledEventMetadata", "GuildScheduledEvent")
+
+
+class GuildScheduledEventMetadata:
+    """
+    Represents guild event entity metadata.
+
+    .. versionadded:: 2.3
+
+    Attributes
+    ----------
+    location: Optional[:class:`str`]
+        Location of the event
```

Comment:
```suggestion
        Location of the event.
        If :attr:`GuildScheduledEvent.entity_type` is :class:`GuildScheduledEventEntityType.external`, this value is not `None`.
```
(not sure if the references resolve properly, haven't tested it)